deploy/buildtools/buildconfig.py

```diff
@@ -1,177 +1,151 @@
+    build_farm_host_dispatcher: BuildFarmHostDispatcher
+
+    def __init__(self, name: str, recipe_config_dict: Dict[str, Any], build_farm_hosts_config_file: Dict[str, Any], build_config_file: BuildConfigFile, launch_time: str) -> None:
+        """
+        Args:
+            name: Name of config i.e. name of `config_build_recipe.yaml` section.
+            recipe_config_dict: `config_build_recipe.yaml` options associated with name.
+            build_farm_hosts_config_file: Parsed representation of `config_build_farm_hosts.yaml` file.
+            build_config_file: Global build config file.
+            launch_time: Time manager was launched.
+        """
         self.name = name
-        self.TARGET_PROJECT = buildconfigdict.get('TARGET_PROJECT')
-        self.DESIGN = buildconfigdict['DESIGN']
-        self.TARGET_CONFIG = buildconfigdict['TARGET_CONFIG']
-        self.PLATFORM_CONFIG = buildconfigdict['PLATFORM_CONFIG']
-        self.instancetype = buildconfigdict['instancetype']
-        self.deploytriplet = buildconfigdict['deploytriplet']
+        self.build_config_file = build_config_file
+
+        self.TARGET_PROJECT = recipe_config_dict.get('TARGET_PROJECT')
+        self.DESIGN = recipe_config_dict['DESIGN']
+        self.TARGET_CONFIG = recipe_config_dict['TARGET_CONFIG']
+        self.deploytriplet = recipe_config_dict['deploy-triplet']
         self.launch_time = launch_time
-        self.launched_instance_object = None
 
-    def __repr__(self):
-        return "BuildConfig obj:\n" + pprint.pformat(vars(self), indent=10)
+        # run platform specific options
+        self.PLATFORM_CONFIG = recipe_config_dict['PLATFORM_CONFIG']
+        self.s3_bucketname = recipe_config_dict['s3-bucket-name']
+        if valid_aws_configure_creds():
+            aws_resource_names_dict = aws_resource_names()
+            if aws_resource_names_dict['s3bucketname'] is not None:
+                # in tutorial mode, special s3 bucket name
+                self.s3_bucketname = aws_resource_names_dict['s3bucketname']
+        self.post_build_hook = recipe_config_dict['post-build-hook']
+
+        # retrieve the build host section
+        self.build_farm_host = recipe_config_dict.get('build-farm', "default-build-farm")
+        build_farm_host_conf_dict = build_farm_hosts_config_file[self.build_farm_host]
+
+        build_farm_host_type = build_farm_host_conf_dict["build-farm-type"]
+        build_farm_host_args = build_farm_host_conf_dict["args"]
+
+        build_farm_host_dispatch_dict = dict([(x.NAME, x.__name__) for x in inheritors(BuildFarmHostDispatcher)])
+
+        build_farm_host_dispatcher_class_name = build_farm_host_dispatch_dict[build_farm_host_type]
+
+        # create dispatcher object using class given and pass args to it
+        self.build_farm_host_dispatcher = getattr(
+            import_module("buildtools.buildfarmhostdispatcher"),
+            build_farm_host_dispatcher_class_name)(self, build_farm_host_args)
```

Comment:
Also, related to `build_farm_host_dispatch_dict` it seemed a little weird to me that it was a `Dict[str, str]` instead of being `Dict[str, type]`.  If you changed it to `Dict[str,type]` you could avoid doing this weird `getattr()` and more-importantly, it will pave the way to being able to have other plugin classes that derive from `BuildFarmHostDispatcher` that don't have to live inside the mainline FireSim `buildtools.buildfarmhostdispatcher` module.

Reply:
Ok fixed.